### PR TITLE
feat(clean): add clean_headers function

### PR DESCRIPTION
**Pull request by atol**
# Description

Implements the `clean_headers()` function to clean column names. Design details are set out in #447.

# How Has This Been Tested?

See snapshots below.

# Snapshots:

## Functionality

clean_headers

## Specify case style

Users can specify the case style using the `case` parameter:

styles_1

styles_2

styles_3

## Replace values

Users can specify values to replace in the column names by passing in a dictionary to the `replace` parameter:

replace

## Remove accents

Accents are automatically removed from column names:

accents_1

Removal of accents can be suppressed by setting `remove_accents=False`:

accents_2

## Rename duplicated column names

Column names duplicated by calling `clean_headers()` are automatically renamed by appending a number to the end of the duplicated column name(s):

duplicate_1

The suffix used to append the number is inferred from the `case` type:

duplicate_2

## Real World Data

An example of `clean_headers()` called on the [FIFA 21 messy, raw dataset for cleaning/ exploring](https://www.kaggle.com/yagunnersya/fifa-21-messy-raw-dataset-for-cleaning-exploring) dataset:

real_world_1

real_world_2

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have already squashed the commits and made the commit message conform to the project standard.
- [ ] I have already marked the commit with "BREAKING CHANGE" or "Fixes #" if needed.
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: dataprep/clean/clean_headers.py [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        return name
+
+    if case in {"snake", "kebab", "camel", "pascal", "const"}:
+        words = _split_strip_string(str(name))
+    else:
+        words = _split_string(str(name))
+
+    if case == "snake":
+        name = "_".join(words).lower()
+    if case == "kebab":
```
Comment (brandonlockhart) [inline dataprep/clean/clean_headers.py:154]: Should this be `elif` instead of repeating `if` statements?

File: dataprep/clean/clean_headers.py [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+        "upper",
+    }
+
+    if case not in case_styles:
+        raise ValueError(
+            f"case {case} is invalid, it needs to be one of {', '.join(c for c in case_styles)}"
+        )
+
+    # Convert to dask
+    df = to_dask(df)
```
Comment (brandonlockhart) [inline dataprep/clean/clean_headers.py:102]: I recommend not going converting to dask. I think the API to access the columns for a Dask and pandas DataFrame is `df.columns`, so I don't think you will need to consider whether a Dask or pandas df is passed.

File: dataprep/clean/clean_headers.py [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+    df.columns = _rename_duplicates(df.columns, case)
+
+    stats = {}
+    # Count the number of changed column names
+    stats["cleaned"] = len(df.columns.astype(str).difference(orig_columns.astype(str)))
+    # Count null column values
+    stats["null"] = len([col for col in list(df.columns) if col in NULL_VALUES])
+
+    with ProgressBar(minimum=1, disable=not progress):
```
Comment (brandonlockhart) [inline dataprep/clean/clean_headers.py:123]: I think you can remove `dask.compute(df)` here, and also the progress bar parameter from `clean_headers`.

**Review by brandonlockhart — COMMENTED** [review]
Very nice PR Alice! I like how you structured the code and your string manipulation looks good to me, and thanks for following the docstring/coding guidelines carefully. I left some small comments below.

I'm wondering about how we're handling null headers. What do you think about creating a default value, eg 'header', and replacing any null headers with the value 'header'? So if we have three null column headers and `case='snake'`, the null column headers would be replaced with 'header', 'header_2', 'header_3'? (I think this is how you handle duplicate headers) 

Let me know what you think.

**Comment by atol** [comment]
Thanks for your comments, Brandon! I agree with your suggestion on how to handle null headers and have implemented it accordingly. Since we now rename null headers, it seems unnecessary to keep track of null values. As such, I have removed that statistic from the cleaning report as well.

null_headers

File: dataprep/clean/clean_headers.py [hunk 4]
```diff
@@ -0,0 +1,246 @@
+"""
+Implement clean_headers function
```
Comment (brandonlockhart) [inline dataprep/clean/clean_headers.py:2]: Can you please take a look at point 1. (Module Docstring) [here](https://github.com/sfu-db/dataprep/wiki/Docstring-Best-Practice#dataprepclean-docstring-recommendations). I'm wondering if we could write a clearer brief description of the functionality in this module, perhaps something like "Clean and standardize column headers." (or whatever you think is best)

**Review by brandonlockhart — CHANGES_REQUESTED** [review]
Thanks for making the revisions, Alice! I just left one more comment.

Also, can you please follow the instructions [here](https://github.com/sfu-db/dataprep/wiki/Docstring-Best-Practice#auto-generating-the-api-reference) and add the modified file to this PR so that your docstrings will be added to the documentation in the next release?

The function works great from my testing though; so after these two changes, I think it can be merged!

**Comment by atol** [comment]
Sorry about that! I wasn't sure whether to include the changes to `dataprep.clean.rst` in this PR or a later one for the docs. I've added the modified file and updated the Module Docstring.

**Comment by brandonlockhart** [comment]
No problem! I think `dataprep.clean.rst` can be added at this stage since the docstrings are ready. And if there's a release before the user guide is added, the user can still learn about this function from the API reference.

**Pull request merged**